src/main/java/com/mparticle/kits/TaplyticsKit.java

```diff
@@ -0,0 +1,300 @@
+        implements
+        KitIntegration.AttributeListener,
+        KitIntegration.EventListener,
+        KitIntegration.CommerceListener,
+        KitIntegration.ApplicationStateListener {
+
+    /**
+     * Option Keys
+     */
+    private static final String API_KEY = "apiKey";
+    private static final String AGGRESSIVE = "TaplyticsOptionAggressive";
+    private static final String USER_ID = "user_id";
+    private static final String EMAIL = "email";
+
+    /**
+     * Event Keys
+     */
+    private static final String EventAppBackground = "appBackground";
+    private static final String EventAppForeground = "appForeground";
+
+    /**
+     * tlOptions get and set methods
+     */
+
+    private static Map<String, Object> tlOptions;
+
+    public static Map<String, Object> getTlOptions() { return tlOptions; }
+
+    public static void setTlOptions(Map<String, Object> options) {
+        tlOptions = options;
+    }
+
+    private HashMap<String, Object> mergeOptions(Map<String, Object> tlOptions, Map<String, Object> configuration) {
+        HashMap<String, Object> merged = new HashMap<>(configuration);
+        for (Map.Entry<String, Object> entry : tlOptions.entrySet()) {
+            merged.put(entry.getKey(), entry.getValue());
+        }
+        return merged;
+    }
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        if (tlOptions == null) {
+            tlOptions = new HashMap<>();
+        }
+        startTaplytics(settings, context);
+        return null;
+    }
+
+    /**
+     * Report Messaging Helper methods
+     */
+
+    private ReportingMessage createReportMessage(String reportMessage) {
+        return new ReportingMessage(this,
+                                    reportMessage,
+                                    System.currentTimeMillis(),
+                                    null);
+    }
+
+    private List<ReportingMessage> createReportingMessages(ReportingMessage report) {
+        return Collections.singletonList(report);
+    }
+
+    private List<ReportingMessage> createReportingMessages(String message) {
+        return Collections.singletonList(createReportMessage(message));
+    }
+
+    /**
+     * Start Taplytics
+     * @param settings
+     * @param context
+     */
+
+    private void startTaplytics(Map<String, String> settings, Context context) {
+        String apiKey = getAPIKey(settings);
+        HashMap<String, Object> options = mergeOptions(getTlOptions(), getOptionsFromConfiguration(settings));
+        options.put("delayedStartTaplytics", true);
+
+        if (options != null) {
+            Taplytics.startTaplytics(context, apiKey, options);
+            return;
+        }
+
+        Taplytics.startTaplytics(context, apiKey);
+    }
+
+    private String getAPIKey(Map<String, String> settings) {
+        final String apiKey = getSettings().get(API_KEY);
+        if (TextUtils.isEmpty(apiKey)) {
```

Comment:
Can you use KitUtils.isEmpty instead? Otherwise unit tests fill failure as TextUtils isn't available without an emulator.